### ISSUE TITLE
Human friendly releases with changelog?

### ISSUE THREAD

**Issue by mahdi1234** [TryQuiet/quiet#908]
I would very much like to track progress of app with having some changelog for the app as such unfortunately https://github.com/TryQuiet/quiet/releases is right now kinda useless.

Would you consider a better one?

**Comment by holmesworcester** [comment]
Thanks for the suggestion and apologies for the slow reply. There are two separate issues here: 

1. Hiding releases of packages like `backend` and `state-manager` from our monorepo's releases page and only showing releases of Quiet for desktop and mobile.
2. Having nice CHANGELOGs for both desktop and mobile

It was also requested that we show the changelog in the respective apps, so that users can see what's new on update. We can create a separate issue for that once this is done.

**Comment by EmiM** [comment]
@holmesworcester 
> Having nice CHANGELOGs for both desktop and mobile

Could you specify what is "nice"? Do you mean regular changelog wich sections like "fix", "feature" grouped by versions OR something more?

**Comment by holmesworcester** [comment]
We could try automatically generated notes. It's based on PR titles, so we'd need to be descriptive with our PR titles, right?

**Comment by holmesworcester** [comment]
@Kacper-RF how about let's try using Github's automated release notes for the 2.0 release. At the stage where you can edit them, take a pass at them and send them to me.

**Comment by siepra** [comment]
We can stop releasing them (I marked them as 'private' at some point) but I remember @leblowl insisting on publishing them anyway

**Comment by holmesworcester** [comment]
@Kacper-RF can you stop releasing them in the next release?

**Comment by Kacper-RF** [comment]
Sure, I might even try to start doing this for alpha releases, the releases page will be much cleaner.

**Comment by siepra** [comment]
All you have to do is to add flag `--no-private` to `lerna version` command

**Comment by EmiM** [comment]
What's the decision here and do we still want to add this within current sprint?

**Comment by holmesworcester** [comment]
I think our proposal is:

1. add flag `--no-private`  to `lerna version` command for a cleaner releases page (kacper already did this once i believe)
2. try using Github's automated release notes
3. edit it heavily if necessary 

For 3, I think our goal should be: 

Everything should be understandable at least to a technical user who does not know our internals, and ideally to a non-technical user)

**Comment by siepra** [comment]
I don't quite get it. If the point is to make changelog understandable and we already moderate changelog manually on develop branch (it's not yet merged into master) then what's the point of populating it with automatic output from GitHub? It'll make us edit it which will only be extra work.

Also, to avoid misunderstanding, your first point is misleading. The person publishing a release should _use_ `--no-private` flag. There's nowhere to _add_ the flag.

**Comment by Kacper-RF** [comment]
Using `--no-private` flag works great.
Tried in today's alpha release:


![Image](https://github.com/TryQuiet/quiet/assets/111343884/1a9b9ec2-57b1-4517-88ec-b909fcace085)

**Comment by holmesworcester** [comment]
As long as it's clear to users and not cluttered, I'm okay with changelog or autogenerated+edited release notes.

If we go the changelog route these should become release notes on the releases page too, so that someone can look at the releases page and easily see what happened in each version.

**Comment by siepra** [comment]
My proposition is to mirror CHANGELOG.md from root directory into individual packages' CHANGELOG.md files.
Here is a raw script for mirroring (it needs adjustments):
```
mirror-changelog.js


const fs = require('fs');
const path = require('path');

// Define the root directory and the packages directory
const rootDir = path.resolve(__dirname, '..');
const packagesDir = path.resolve(rootDir, 'packages');

// Read the root CHANGELOG.md file
const rootChangelog = fs.readFileSync(path.resolve(rootDir, 'CHANGELOG.md'), 'utf8');

// Get a list of all packages
const packages = fs.readdirSync(packagesDir);

// Copy the root CHANGELOG.md file to each package
packages.forEach((package) => {
  fs.writeFileSync(path.resolve(packagesDir, package, 'CHANGELOG.md'), rootChangelog);
});
```
and here is the proposition for publishing script for roots' `package.json`:
```
  "scripts": {
    "publish": "lerna version $npm_config_release --no-private"
    "postpublish": "node copy-changelog.js"
  }
```
and the example of an usage:
```
npm run publish --release=minor
```

**Comment by holmesworcester** [comment]
Can we edit release notes after release?

If not, let's review quickly in Slack before releasing.

**Comment by holmesworcester** [comment]
Notes: @Kacper-RF  will add this script.

**Comment by Kacper-RF** [comment]
https://github.com/TryQuiet/quiet/pull/2124
